app/routes/api/release.js

```diff
@@ -187,7 +197,30 @@ router.get('/:slugOrBundleId', cache, function(req, res) {
       token: req.query.token,
       debug: req.query.debug
     },
-    response.bind(res)
+    function(err, release) {
+      if (err === null && release !== null) {
+        // if there's no error and a release was found
+        return res.send({ success: true, data: release });
+      } else if (err === null && release === null) {
+        // no release was found, so it's a 404
+        log.warn(`No release found for slug "${req.params.slugOrBundleId}"`);
+        return res.status(404).send({ success: false, data: null });
+      }
+
+      if (err === 'Invalid game slug') {
+        // If the slug doesn't exist, it's a 404
+        log.warn(`Invalid game slug "${req.params.slugOrBundleId}"`);
+        return res.status(404).send({ success: false, error: err });
+      } else if (err === 'Token is required') {
+        // If the request was for a dev game without a token, it's a 403
+        log.warn(`Request for game "${req.params.slugOrBundleId}" without token`);
+        return res.status(403).send({ success: false, error: err });
+      } else {
+        // Otherwise, we don't know what it is so it's probably a 500
```

Comment:
Shouldn't the default error be a 400, not a 500? Since it seems more likely that the request was bad than the server errored? I guess you could make the argument that this was an error thrown somewhere up the chain, so it could very easily be a real 500? Idk...

Reply:
IMO my reasoning is that if we don't know where the error came from, we should assume that it's our fault we can't really "blame the user" in this case. Moreover, triggering a 500 here will give us visibility into errors with unknown sources since our monitoring solutions should pick this up.